core/blockchain.go

```diff
@@ -3167,3 +3187,52 @@ func isUnrecoverableErr(err error) bool {
 	isTooManyOpenFiles := strings.Contains(err.Error(), tooManyOpenFilesErrStr)
 	return isLeveldbErr && !isTooManyOpenFiles
 }
+
+// PruneValidatorWrappersAtBlock takes a block number which must be already available
+// re-creates the state, removes the validator wrappers and writes the state to disk
+func (bc *BlockChain) PruneValidatorWrappersAtBlock(number uint64, checked bool) error {
+	if number == 0 {
+		return errors.New("Cannot trim rewards for genesis block")
+	}
+	block := bc.GetBlockByNumber(number)
+	if block != nil {
+		clearValidatorWrappers := checked || // if already checked, skip
+			(!bc.chainConfig.IsAggregatedRewardEpoch(block.Epoch()) && // before aggregated reward epoch
+				number%chain.RewardFrequency != chain.RewardFrequency-1 && // keep every 64th block
```

Comment:
We actually don't need to keep the wrapper every 64 block. We can remove all the history.